test/cli/run.spec.ts

```diff
@@ -0,0 +1,388 @@
+// This is typescript so that it only runs in node contexts, not on the web
+
+import * as peggy from "../../lib/peg.js";
+import fs from "fs";
+import path from "path";
+import { spawn } from "child_process";
+
+type Options = {
+  args?: string[];
+  encoding?: BufferEncoding;
+  env?: Record<string, string>;
+  stdin?: string | Buffer;
+};
+
+/** Execution failed */
+class ExecError extends Error {
+  /** Result error code, always non-zero */
+  code: number;
+
+  /** Stdout as a Buffer */
+  buf: Buffer;
+
+  /** Stdout as a string, decoded with opts.encoding */
+  str: string;
+
+  constructor(message: string, code: number, buf: Buffer, str: string) {
+    super(`${message}: error code "${code}"
+${str}`);
+    this.name = "ExecError";
+    Object.setPrototypeOf(this, ExecError.prototype);
+    this.code = code;
+    this.buf = buf;
+    this.str = str;
+  }
+}
+
+function exec(opts: Options = {}) {
+  opts = {
+    args: [],
+    encoding: "utf8",
+    env: {},
+    ...opts,
+  };
+  return new Promise((resolve, reject) => {
+    const bin = path.join(__dirname, "..", "..", "bin", "peggy.js");
+    const env = {
+      ...process.env,
+      ...opts.env,
+    };
+    const c = spawn(bin, opts.args, {
```

Comment:
Tests on windows failed because this line executes script using shebang magic, which not work under Windows. This should work:
```suggestion
    if (opts.args !== undefined) {// strange, just check if (opts.args) is not enough
      opts.args.unshift(bin);
    } else {
      opts.args = [bin];
    }
    const c = spawn("node", opts.args, {
```

See https://stackoverflow.com/questions/43419893/how-do-i-fix-error-spawn-unknown-with-node-js-v7-8-0-on-windows-10

Reply:
I did this if `process.platform === "win32"`, which lets me test both ways.

Reply:
Ok